component/src/main/java/io/siddhi/extension/io/file/FileHandler.java

```diff
@@ -0,0 +1,409 @@
+            sourceEventListener.onEvent
+                    (getFileHandlerEvent(file, listeningFileUri, STATUS_DONE), null);
+        }
+
+        @Override
+        public void onFileChange(final File file) {
+            log.info(file.getAbsolutePath() + " was modified.");
+            synchronized (this) {
+                if (!initialMap.containsKey(file.getAbsolutePath())) {
+                    initialMap.put(file.getAbsolutePath(), file.length());
+                    sourceEventListener.onEvent
+                            (getFileHandlerEvent(file, listeningFileUri, STATUS_PROCESS), null);
+                    SubThread subThread = new SubThread(file);
+                    siddhiAppContext.getScheduledExecutorService().schedule(subThread,
+                            monitoringInterval, TimeUnit.MILLISECONDS);
+                    threadList.add(subThread);
+                }
+            }
+        }
+
+        @Override
+        public void onFileDelete(final File file) {
+            log.debug(file.getAbsolutePath() + " was deleted.");
+            sourceEventListener.onEvent
+                    (getFileHandlerEvent(file, listeningFileUri, STATUS_DONE), null);
+        }
+
+        @Override
+        public void onStop(final FileAlterationObserver observer) {
+        }
+    }
+
+    public void initiateFileAlterationObserver() {
+        FileAlterationObserver observer = new FileAlterationObserver(listeningDirUri);
+        observer.addListener(new FileAlterationImpl());
+        monitor = new FileAlterationMonitor(monitoringInterval);
+        monitor.addObserver(observer);
+        try {
+            monitor.start();
+            log.info("Directory monitoring has been started for folder : " + uri + " .");
+        } catch (Exception e) {
+            throw new SiddhiAppRuntimeException("Exception occurred when starting server to monitor " + uri + ".", e);
+        }
+    }
+
+    @Override
+    public void disconnect() {
+
+        if (monitor != null) {
+            try {
+                monitor.stop();
+                initialMap.clear();
+                log.info("Directory monitoring has been stopped for folder : " + uri + " .");
+            } catch (Exception e) {
+                log.error("Exception occurred when stopping server  while monitoring " + uri + " .", e);
+            }
+        }
+    }
+
+    @Override
+    public void destroy() {
+    }
+
+    @Override
+    public void pause() {
+        if (monitor != null) {
+            threadList.forEach(SubThread::pause);
+            log.info("Directory monitoring has been paused for folder : " + uri + " .");
+        }
+    }
+
+    @Override
+    public void resume() {
+        if (monitor != null) {
+            threadList.forEach(SubThread::resume);
+            log.info("Directory monitoring has been resumed for folder : " + uri + " .");
+        }
+    }
+
+    /**
+     * Implementation of SubThread
+     */
+    public class SubThread implements Runnable {
```

Comment:
Lets change the name to a meaningful name which describes what it does

Reply:
SubThread class is removed.